apps/openmw/mwrender/sky.cpp

```diff
@@ -1730,7 +1747,8 @@ void SkyManager::setWeather(const WeatherResult& weather)
                 mParticleNode = new osg::PositionAttitudeTransform;
                 mParticleNode->addCullCallback(mUnderwaterSwitch);
                 mParticleNode->setNodeMask(Mask_WeatherParticles);
-                mRootNode->addChild(mParticleNode);
+                mParticleNode->addCullCallback(new SceneUtil::LightListCallback);
```

Comment:
Stuff that uhh enables lighting on weather particle systems I guess should be put to a separate PR, or we'll never get particle system fading (the core issue and an important regression) fixed. I'm also confused as to why is all the dancing around two root nodes necessary.

Reply:
dancing around two root nodes is required cause sky is not shadable (or in hardcoded manner i guess) but particles are